src/main/java/com/iexec/core/workflow/ReplicateWorkflow.java

```diff
@@ -17,40 +38,20 @@ private ReplicateWorkflow() {
         addTransition(RUNNING, APP_DOWNLOADING);
 
         // app
-        addTransition(APP_DOWNLOADING, toList(APP_DOWNLOADED, APP_DOWNLOAD_FAILED));
-
-        addTransition(APP_DOWNLOAD_FAILED, toList(
-                // DATA_DOWNLOADING,
-                CANT_CONTRIBUTE_SINCE_STAKE_TOO_LOW,
-                CANT_CONTRIBUTE_SINCE_TASK_NOT_ACTIVE,
-                CANT_CONTRIBUTE_SINCE_AFTER_DEADLINE,
-                CANT_CONTRIBUTE_SINCE_CONTRIBUTION_ALREADY_SET,
-                CAN_CONTRIBUTE));
-
+        addTransition(APP_DOWNLOADING, APP_DOWNLOADED_AND_APP_DOWNLOAD_FAILED_LIST);
+        addTransition(APP_DOWNLOAD_FAILED, CAN_AND_CANT_CONTRIBUTE_LIST);
         addTransition(APP_DOWNLOADED, DATA_DOWNLOADING);
 
         // data
-        addTransition(DATA_DOWNLOADING, toList(DATA_DOWNLOADED, DATA_DOWNLOAD_FAILED));
-
-        addTransition(DATA_DOWNLOAD_FAILED, toList(
-                // COMPUTING,
-                CANT_CONTRIBUTE_SINCE_STAKE_TOO_LOW,
-                CANT_CONTRIBUTE_SINCE_TASK_NOT_ACTIVE,
-                CANT_CONTRIBUTE_SINCE_AFTER_DEADLINE,
-                CANT_CONTRIBUTE_SINCE_CONTRIBUTION_ALREADY_SET,
-                CAN_CONTRIBUTE));
-
+        addTransition(DATA_DOWNLOADING, DATA_DOWNLOADED_AND_DATA_DOWNLOAD_FAILED_LIST);
+        addTransition(DATA_DOWNLOAD_FAILED, CAN_AND_CANT_CONTRIBUTE_LIST);
         addTransition(DATA_DOWNLOADED, COMPUTING);
 
         // computation
-        addTransition(COMPUTING, toList(COMPUTED, COMPUTE_FAILED));
-
-        addTransition(COMPUTED, toList(
-                CANT_CONTRIBUTE_SINCE_STAKE_TOO_LOW,
-                CANT_CONTRIBUTE_SINCE_TASK_NOT_ACTIVE,
-                CANT_CONTRIBUTE_SINCE_AFTER_DEADLINE,
-                CANT_CONTRIBUTE_SINCE_CONTRIBUTION_ALREADY_SET,
-                CAN_CONTRIBUTE));
+        addTransition(COMPUTING, COMPUTED_AND_COMPUTE_FAILED_LIST);
+        for (ReplicateStatus from : COMPUTED_AND_COMPUTE_FAILED_LIST) {
```

Comment:
Maybe this would be more readable instead of a loop:
`addTransition(COMPUTE_FAILED, CAN_AND_CANT_CONTRIBUTE_LIST);
addTransition(COMPUTED, CAN_AND_CANT_CONTRIBUTE_LIST);`